hs_file_types/views.py

```diff
@@ -738,6 +738,12 @@ def add_keyword_metadata(request, hs_file_type, file_type_id, **kwargs):
                               'element_name': 'keyword', 'message': "Permission denied"}
         return JsonResponse(ajax_response_data, status=status.HTTP_200_OK)
 
+    if resource.raccess.published:
+        ajax_response_data = {'status': 'error', 'logical_file_type': logical_file.type_name(),
+                              'element_name': 'keyword',
+                              'message': "Editing of keywords is not allowed for a published resource"}
+        return JsonResponse(ajax_response_data, status=status.HTTP_200_OK)
```

Comment:
A 200 OK with an error status is incorrect.  I imagine you're just using the same status code as the not authorized response above to stay consistent for the frontend.  I don't expect action here, but if you see a straightforward way to update the responses to 403 for all of these, it'd be nice to break this pattern that was set earlier.

Reply:
You are right. The problem is the current JS code base doesn't handle error code well and depends mostly on the the 'status' response to handle error condition. If it was only backend code I could have fixed it.